packages/arro3-compute/meta.yaml

```diff
@@ -0,0 +1,20 @@
+package:
+  name: arro3-compute
+  version: 0.3.0
+  top-level:
+    - "arro3.compute"
```

Comment:
CI is failing because top-level isn't supposed to contain a dot. Either these should be `top-level: [arro3]` or we need to fix `find_imports`. @ryanking13 perhaps it would make sense to change `find_imports` to support this:

```patch
+def _add_prefixes(s: set[str], mod: str) -> None:
+    [current, *rest] = mod.split(".")
+    s.add(current)
+    for part in rest:
+        current += f".{part}"
+        s.add(current)
+
+
 def find_imports(source: str) -> list[str]:
     """
     Finds the imports in a Python source code string
@@ -635,17 +643,17 @@ def find_imports(source: str) -> list[str]:
         mod = ast.parse(source)
     except SyntaxError:
         return []
-    imports = set()
+    imports: set[str] = set()
     for node in ast.walk(mod):
         if isinstance(node, ast.Import):
             for name in node.names:
                 node_name = name.name
-                imports.add(node_name.split(".")[0])
+                _add_prefixes(imports, node_name)
         elif isinstance(node, ast.ImportFrom):
             module_name = node.module
             if module_name is None:
                 continue
-            imports.add(module_name.split(".")[0])
+            _add_prefixes(imports, module_name)
     return list(sorted(imports))
```

Reply:
I really like namespace packaging but it does tend to break tools 😅 

> Either these should be top-level: [arro3]

For a namespace module `arro3.core`, `import arro3` will do _nothing_ because there's no `__init__.py` file in the `arro3` directory. It's not until you reach `arro3/core/__init__.py` that there's anything to import.

Reply:
Yes, I think it makes sense to change find_imports. Changing `top-level: [arro3]` would result in importing `arro3` loading all three arro3 subpackages, which might not be the intended behavior. I am not sure it is a good practice to locate three different packages in a same top-level directory though.

Reply:
> Changing `top-level: [arro3]` would result in importing `arro3` loading all three arro3 subpackages

No, with [namespace packaging](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/), calling `import arro3` does nothing. No code is loaded until importing one of the submodules:

```
In [1]: import arro3

In [2]: arro3.core.Array
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[2], line 1
----> 1 arro3.core.Array

AttributeError: module 'arro3' has no attribute 'core'

In [3]: import arro3.core

In [4]: arro3.core.Array
Out[4]: arro3.core._core.Array
```